### PR TITLE
GL objects can be acquired and released

### PR DESCRIPTION
**Pull request by Kerosene2000**

### CODE REVIEW

File: src/standard/buffer.rs [hunk 1]
```diff
@@ -454,17 +488,48 @@ impl<T: OclPrm> Buffer<T> {
...
+    ///
+    /// ## Errors
+    ///
+    /// Don't forget to `.cmd().gl_acquire().enq()` before using it and
+    /// `.cmd().gl_release().enq()` after.
+    ///
+    /// See the [`BufferCmd` docs](/ocl/ocl/build/struct.BufferCmd.html)
+    /// for more info.
+    ///
+    // FIXME: need GLuint type in place of gl_object ! did I import all types from cl_h.rs ?
```
Comment (Kerosene2000) [inline src/standard/buffer.rs]: Have you any idea to fix this ? Did I need to import all cl_h.rs types ?

Comment (c0gent) [inline src/standard/buffer.rs]: I believe I do but I need you to do something first...

I noticed you've fixed files where I had incorrect whitespace. Would you mind please making a separate commit or pull request just for white space only. I'll merge that first then we can have a look at your issue more clearly.

Thanks.

**Comment by c0gent** [comment]
Please make whitespace a separate commit.

**Comment by c0gent** [comment]
As soon as you finish that, the type you need is now available at `ocl::aliases::ClGlUint` as of the latest commit.

File: src/standard/buffer.rs (hunk 1 above)
Comment (Kerosene2000) [inline src/standard/buffer.rs]: I will do that, I am sorry for that, my text editor configuration does this itself...

**Comment by c0gent** [comment]
Great feature addition. I've been wanting to do this for a while.

A few things I'd like you to change:

* I appreciate you creating separate commits for the whitespace changes but I think now it would be better to move them into a separate pull request entirely. I know this is a hassle for you but this pull request has become too noisy. I'd really appreciate if you could:
  1. Re-fork the master branch into a new repo which you will use to send new pull requests.
  2. Make all the whitespace changes to files.
  3. Commit and send pull request.
  4. Copy and paste your code changes and additions.
  5. Commit and send pull request.
* Please move the `clEnqueueAcquireGLObjects` and `clEnqueueReleaseGLObjects` functions into the new `ffi::cl_gl_h.rs` file. I'm planning to move the entire `cl_h.rs` file into that module soon but I would like to keep their contents separate so that any OpenCL version changes are easier to integrate.

I won't bother you with creating tests, after you finish the above changes I will create some tests myself unless you had something in mind already. Let me know.

Thanks :)

**Pull request closed without merging**